homeassistant/components/entur_public_transport/sensor.py

```diff
@@ -170,33 +195,51 @@ def icon(self) -> str:
         """Icon to use in the frontend."""
         return self._icon
 
-    def update(self) -> None:
+    async def async_update(self) -> None:
         """Get the latest data and update the states."""
-        from enturclient.consts import (
-            ATTR, ATTR_EXPECTED_AT, ATTR_NEXT_UP_AT, CONF_LOCATION,
-            CONF_LATITUDE as LAT, CONF_LONGITUDE as LONG, CONF_TRANSPORT_MODE)
-
-        self.api.update()
+        await self.api.async_update()
 
         data = self.api.get_stop_info(self._stop)
-        if data is not None and ATTR in data:
-            attrs = data[ATTR]
-            self._attributes.update(attrs)
-
-            if ATTR_NEXT_UP_AT in attrs:
-                self._attributes[ATTR_NEXT_UP_IN] = \
-                    due_in_minutes(attrs[ATTR_NEXT_UP_AT])
-
-            if CONF_LOCATION in data and self._show_on_map:
-                self._attributes[CONF_LATITUDE] = data[CONF_LOCATION][LAT]
-                self._attributes[CONF_LONGITUDE] = data[CONF_LOCATION][LONG]
-
-            if ATTR_EXPECTED_AT in attrs:
-                self._state = due_in_minutes(attrs[ATTR_EXPECTED_AT])
-            else:
-                self._state = None
-
-            self._icon = ICONS.get(
-                data[CONF_TRANSPORT_MODE], ICONS[DEFAULT_ICON_KEY])
-        else:
+
+        if data is None:
+            self._state = None
+            return
+
```

Comment:
Are we here risking that old attributes never gets removed?

self. _attributes = {}
 Is that needed to clear the old attributes?

Will len(calls)  always be the same?

Reply:
Clearing of the attributes should have been added now. Don't think it would leave any attributes behind, but better to be safe than sorry.

Reply:
Then you also have to move 
```
            ATTR_ATTRIBUTION: ATTRIBUTION,
            ATTR_STOP_ID: self._stop,
```
`device_state_attributes()
`

Reply:
Good catch! Will fix

Reply:
Hopefully all better now